core/trino-main/src/main/java/io/trino/execution/SqlTaskManager.java

```diff
@@ -494,7 +519,13 @@ private TaskInfo doUpdateTask(
             }
         }
 
-        fragment.ifPresent(planFragment -> connectorServicesProvider.ensureCatalogsLoaded(session, planFragment.getActiveCatalogs()));
+        if (fragment.isPresent()) {
```

Comment:
Could this stay as a closure? Also, should we only call `getActiveCatalogs()` once, to ensure a consistent set?
```java
fragment.ifPresent(planFragment -> {
    List<CatalogProperties> activeCatalogs = planFragment.getActiveCatalogs();
    Set<CatalogHandle> catalogs = activeCatalogs.stream()
            .map(CatalogProperties::getCatalogHandle)
            .collect(toImmutableSet()))) {
    if (sqlTask.setCatalogs(catalogs)) {
        connectorServicesProvider.ensureCatalogsLoaded(session, activeCatalogs);
    }
});